sandy/core/cov.py

```diff
@@ -1605,87 +1531,30 @@ def get_L(self, rows=None):
         --------
         >>> a = np.array([[4, 12, -16], [12, 37, -43], [-16, -43, 98]])
         >>> sandy.CategoryCov(a).get_L()
-                      0	          1	          2
-        0	2.00000e+00	0.00000e+00	0.00000e+00
-        1	6.00000e+00	1.00000e+00	0.00000e+00
-        2  -8.00000e+00	5.00000e+00	3.00000e+00
-
-        >>> sandy.CategoryCov(a).get_L(rows=1)
-                      0	          1	          2
-        0	2.00000e+00	0.00000e+00	0.00000e+00
-        1	6.00000e+00	1.00000e+00	0.00000e+00
-        2  -8.00000e+00	5.00000e+00	3.00000e+00
-
-        >>> sandy.CategoryCov([[1, -2],[-2, 3]]).get_L()
-                       0   	          1
-        0	 1.08204e+00	0.00000e+00
-        1	-1.75078e+00	0.00000e+00
-
-        >>> sandy.CategoryCov([[1, -2],[-2, 3]]).get_L(rows=1)
-                       0   	          1
-        0	 1.08204e+00	0.00000e+00
-        1	-1.75078e+00	0.00000e+00
+                       0	          1	          2
+        0	-2.00000e+00	0.00000e+00	0.00000e+00
+        1	-6.00000e+00	1.00000e+00	0.00000e+00
+        2	 8.00000e+00	5.00000e+00	3.00000e+00
         """
         index = self.data.index
         columns = self.data.columns
-        # Ensure matrix is positive define:
-        values = self.to_positive(rows=rows).data.values
-        rows_ = values.shape[0] if rows is None else rows
-        L = sparse_tables_cholesky(values, rows=rows_)
+        # Reduces the size of the matrix, erasing the zero values
+        nonzero_idxs, cov_reduced = reduce_size(self.data)
+        # Obtain the eigenvalues and eigenvectors:
+        E, V = sandy.CategoryCov(cov_reduced).eig(tolerance=tolerance)
+        E = sandy.CategoryCov.from_var(np.sqrt(E)).data.values
+        # Construct the matrix:
+        rows_ = cov_reduced.shape[0] if rows is None else rows
+        A = sandy.cov.sparse_tables_dot(V, E, rows=rows_).T.toarray()
+        # QR decomposition:
+        Q, R = scipy.linalg.qr(A)
+        L_redu = R.T
+        # Original size
+        L = restore_size(nonzero_idxs, L_redu, len(self.data)).values
+        if threshold is not None:

```

Comment:
I am not sure we need a threshold...are we ever going to use it?
We will not implement it in the sampling pipeline.
And if it is just "in case", then any API user can make this simple truncation...